float64 ox
float64 oy
float64 oz
float64 oR
float64 oP
float64 oY

